peer-to-peer/0077.md

```diff
@@ -0,0 +1,236 @@
+    deactivate Receiver Wallet
+    deactivate Receiver
+    deactivate Receiver
+```
+
+### Making a payment
+
+For making a payment we can leverage the ["BEEF Transaction" capability](../payments/0070.md).
+Thanks to the confirmation of the counterparty public key and the ability to create a shared secret,
+we are now able to create a new universe of addresses dedicated to the counterparty.
+Therefore, we are able to omit usage of [paymentDestination capability](https://bsvalias.org/04-01-basic-address-resolution.html) 
+or [P2P Payment Destination capability](../payments/0028.md).
+
+To achieve high flexibility in generation of stable set of addresses we need to define the following values:
+1. shared secret which is calculated by multiplying sender private key and a confirmed public key of the receiver
+2. reference - a random (or at least unique in context of transactions for receiver) value used to reference the payment
+3. idx - transaction output index for which he is calculating an address
+
+The detailed flow of making the payment for the confirmed contact is shown in the following sequence diagram:
+
+```mermaid
+sequenceDiagram
+    actor Sender
+    participant Sender Wallet
+    participant Sender BUX
+    participant Receiver BUX
+    
+    Sender -->> Sender Wallet: I want to make a payment
+    activate Sender Wallet
+    Note over Sender,Sender BUX: .... ℹ️ Preparation (unlocking) of the inputs for the transactions is presented in the separate diagram...
+    Sender Wallet ->> Sender Wallet: Get public key of contact (recPub)
+    Sender Wallet ->> Sender Wallet: Calculate shared secret (SharedSec)
+    Note right of Sender Wallet: SharedSec = masterPrivateKey * recPub
+    Sender Wallet ->> Sender Wallet: Generate reference (reference)
+
+    loop Every output for receiver
+        Sender Wallet ->> Sender Wallet: get index of output (idx)
+        Sender Wallet ->> Sender Wallet: Calculate HMAC of the shared secret, using reference and idx as key
+        Note right of Sender Wallet: HMAC(message=SharedSec, key= {reference}-{idx})
+        Sender Wallet ->> Sender Wallet: Calculate big number from HMAC modulo N (hn)
+        Sender Wallet ->> Sender Wallet: Calculate dedicated Public Key (dPk[i])
+        Note right of Sender Wallet: dPk[i] = hn * G + recPub
+    end
+
+    Sender Wallet->>Sender BUX: Send transaction with dedicated keys (dPk[]) and reference
+    activate Sender BUX
+    activate Sender Wallet
+    Sender BUX->>Sender BUX: Prepare transaction using dedicated keys (using dPk[i] for proper output)
+    Sender BUX->Sender BUX: Paymail host discovery
+    Sender BUX->Receiver BUX: Paymail capability discovery
+    activate Receiver BUX
+    Sender BUX->Receiver BUX: Get endpoint from BEEF tx capability
+    Sender BUX->>Receiver BUX: Send BEEF
+    Note over Sender BUX,Receiver BUX: reference generated by wallet<br/> should be used in a request
+    Receiver BUX->>Receiver BUX: Make SPV
```

Comment:
This whole thing is perfect except this one line.

If we run SPV here we know that the transactions is valid in the broad sense, but we cannot know which outputs belong to the recipient, and therefore do not know the payment amount.

Just noting here for further discussion.

Reply:
Let's just publish this and come back to fix anything which needs to be updated?